app/helpers/feed_helper.rb

```diff
@@ -14,4 +15,75 @@ def cached_feed(url)
   def feed_entry_thumbnail_url(entry)
     FeedEntryImage.new(entry).thumbnail_url
   end
+
+  def tumblr_feed_content(landing_page, options = {})
+    page = options[:page] || 1
+    per_page = options[:per_page] || 20
+
+    puts "landing_page.feeds.tumblr: #{landing_page.feeds.tumblr.inspect}"
+    feed = landing_page.feeds.tumblr.first
+
+    return nil unless feed
+
+    items = feed_items_for(feed, options)
+
+    return nil if items.blank?
+
+    paginated_items = Kaminari.paginate_array(items).page(page).per(per_page)
+    {
+      title: 'Tumblr',
+      tumblr_url: Cache::FeedJob::URLS[:custom][feed.slug.to_sym].sub('/rss', ''),
```

Comment:
`Cache::FeedJob::URLS[:custom][feed.slug.to_sym]` can be replaced with `feed.url`